lib/puppet/provider/ecs_service/v2.rb

```diff
@@ -0,0 +1,196 @@
+          deployment_configuration: deserialize_deployment_configuration(service.deployment_configuration),
+          cluster: cluster_name,
+        }
+
+        unless service.role_arn.is_a? NilClass
+          ecs_service[:role] = service.role_arn.match(/^(.*)\/([_\w\d-]+)$/)[2]
+          ecs_service[:role_arn] =  service.role_arn
+        end
+
+        new(ecs_service)
+
+      end
+    end
+
+    results.flatten.select {|i| i }
+  end
+
+  read_only(:load_balancers, :role_arn, :role)
+
+  def self.prefetch(resources)
+    instances.each do |prov|
+      # Skipped instances return a nil object, check here for sanity
+      next if prov.is_a? NilClass
+      if resource = resources[prov.name]
+        resource.provider = prov
+      end
+    end
+  end
+
+  def self.deserialize_load_balancers(load_balancers)
+    load_balancers.collect do |lb|
+      normalize_values({
+        load_balancer_name: lb.load_balancer_name,
+        container_name: lb.container_name,
+        container_port: lb.container_port
+      })
+    end
+  end
+
+  def self.deserialize_deployment_configuration(deployment_configuration)
+    normalize_hash({
+      maximum_percent: deployment_configuration.maximum_percent,
+      minimum_healthy_percent: deployment_configuration.minimum_healthy_percent,
+    })
+  end
+
+  def deployment_configuration=(value)
+    @property_flush[:deployment_configuration] = value
+  end
+
+  def desired_count=(value)
+    @property_flush[:desired_count] = value
+  end
+
+  def task_definition=(value)
+    @property_flush[:task_definition] = value
+  end
+
+  def exists?
+    Puppet.debug("Checking if ECS service #{name} exists")
+    @property_hash[:ensure] == :present
+  end
+
+  def create
+    Puppet.debug("Creating ecs_service #{resource[:name]}")
+
+    ecs_service = {
+      service_name: resource[:name],
+      desired_count: resource[:desired_count],
+      task_definition: resource[:task_definition],
+      cluster: resource[:cluster],
+    }
+
+    if resource[:load_balancers]
+      ecs_service[:load_balancers] = resource[:load_balancers]
+    end
+
+    if resource[:role]
+      ecs_service[:role] = resource[:role]
+    end
+
+    if resource[:deployment_configuration]
+      ecs_service[:deployment_configuration] = resource[:deployment_configuration]
+    end
+
+    ecs_client.create_service(ecs_service)
+    @property_hash[:ensure] = :present
+  end
+
+  def destroy
+    Puppet.debug("Destroying ecs_service #{@property_hash[:name]}")
+
+    ecs_client.delete_service({
+      service: @property_hash[:name],
+      cluster: @property_hash[:cluster]
+    })
+    @property_hash[:ensure] = :absent
+  end
+
+  def flush
+    Puppet.debug("Flushing ECS service for #{@property_hash[:name]}")
+
+    if @property_flush.keys.size > 0
+      service_def = {
+        service: @property_hash[:name],
+        cluster: @property_hash[:cluster],
+      }
+
+      if @property_flush.keys.include? :deployment_configuration
```

Comment:
Please change all `Hash.keys.include? :x` to just `Hash[:x]`, which will return `nil` (which evaluates as `false`) if the key is not in the hash.